specification.yaml

```diff
@@ -0,0 +1,82 @@
+openapi: 3.0.3
+info:
+  description: "An OAuth OpenID Connect provider integration as a service"
+  version: "1.0.20"
+  title: "Gatekeeper"
+  contact:
+    name: "Julius Pedersen"
+    email: "developerportal@oslo.kommune.no"
+tags:
+- name: "REST"
+  description: "Entrypoints designed to be used programmatically"
+- name: "redirects"
+  description: "Entrypoints designed to be navigated to"
+paths:
+  /userinfo:
+    get:
+      tags:
+      - REST
+      summary: "Acquire information about the user"
+      description: "Mirrors the authentication providers /userinfo entrypoint. Returns information about the user"
+      parameters:
+      - in: "cookie"
+        name: "access_token"
+        required: true
+        description: "Access token representing a user session"
+        content:
+          "text": {}
+      responses:
+        200:
+          description: Object containing user info as per result from authentication provider
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/UserinfoResponse"
+        401:
+          description: "Missing or invalid access token"
+  /logout:
+    post:
+      tags:
+      - REST
+      summary: Terminates a user session
+      description: Used to invalidate a user session
+      parameters:
+      - in: "cookie"
+        name: "access_token"
+        required: true
+        description: "Access token representing a user session"
+        content:
+          "text": {}
+      responses:
+        200:
+          description: "OK"
+        401:
+          description: "No valid session to log out"
+
+  /login:
+    get:
+      tags:
+      - redirects
+      summary: "Initiate a user session"
+      description: "Generates required configuration and redirects to the authentication provider"
+      parameters:
+      - in: query
+        name: redirect
+        required: true
+        description: Full URL to redirect back to on successful login
+        content:
+          "url": {}
+      responses:
+        302:
+          description: Will redirect the user back to the url specified in the redirect query parameter
```

Comment:
Isn't the response a 302 to the auth provider?

The redirect to the query parameter is the final redirect from the auth provider on a successful login.

Reply:
You're right. Since this is a browser centric entrypoint I was unsure wether to describe the actual response or what the frontend developer / user can expect. Landed on the latter because that way the spec helps convey the usage of the Gatekeeper. 

Explanation try #2:
The API specification should help developers integrate with the API. Since this is an entrypoint used by frontends, I thought it made more sense documenting what the final response/action was, instead of some internal nonsense details not relevant for integration. 

@yngvark do you disagree? I could try to document both

Reply:
In that case I would document both, to avoid surprises / confusion.

Reply:
@yngvark please review the change

Reply:
Looks good :+1: